service/src/main/java/io/camunda/service/JobServices.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.service;
+
+import io.camunda.search.clients.CamundaSearchClient;
+import io.camunda.service.security.auth.Authentication;
+import io.camunda.service.transformers.ServiceTransformers;
+import io.camunda.zeebe.broker.client.api.BrokerClient;
+import io.camunda.zeebe.gateway.impl.broker.request.BrokerActivateJobsRequest;
+import io.camunda.zeebe.gateway.impl.job.ActivateJobsHandler;
+import io.camunda.zeebe.gateway.impl.job.ResponseObserver;
+import java.util.List;
+import java.util.function.Consumer;
+
+public final class JobServices<T> extends ApiServices<JobServices<T>> {
+
+  private final ActivateJobsHandler<T> activateJobsHandler;
+
+  public JobServices(
+      final BrokerClient brokerClient,
+      final ActivateJobsHandler<T> activateJobsHandler,
+      final CamundaSearchClient dataStoreClient) {
+    this(brokerClient, activateJobsHandler, dataStoreClient, null, null);
+  }
+
+  public JobServices(
+      final BrokerClient brokerClient,
+      final ActivateJobsHandler<T> activateJobsHandler,
+      final CamundaSearchClient searchClient,
+      final ServiceTransformers transformers,
+      final Authentication authentication) {
+    super(brokerClient, searchClient, transformers, authentication);
+    this.activateJobsHandler = activateJobsHandler;
+  }
+
+  @Override
+  public JobServices<T> withAuthentication(final Authentication authentication) {
+    return new JobServices<>(
+        brokerClient, activateJobsHandler, searchClient, transformers, authentication);
+  }
+
+  public void activateJobs(
+      final ActivateJobsRequest request,
+      final ResponseObserver<T> responseObserver,
+      final Consumer<Runnable> cancelationHandlerConsumer) {
+    final var brokerRequest =
+        new BrokerActivateJobsRequest(request.type())
+            .setMaxJobsToActivate(request.maxJobsToActivate())
+            .setTenantIds(request.tenantIds())
+            .setTimeout(request.timeout())
+            .setWorker(request.worker())
+            .setVariables(request.fetchVariable());
+    activateJobsHandler.activateJobs(
+        brokerRequest, responseObserver, cancelationHandlerConsumer, request.requestTimeout());
+  }
+
+  public record ActivateJobsRequest(
+      String type,
+      int maxJobsToActivate,
+      List<String> tenantIds,
+      long timeout,
+      String worker,
+      List<String> fetchVariable,
+      long requestTimeout) {}
+}
```

Comment:
💭 I was getting some yellow flags when I saw this `record` since it is another "DTO" class that we'll need to update every time we extend a certain API. And it's not as flexible as the `Broker*Request` classes. But I guess we have the `service` layer now to support us with flexibility.

Reply:
I see your point, yeah. I was also wondering about it. But then again, this is just a convenience for the service method parameters, ultimately. We could alternatively add all those parameters to the service method signature. That would just simply be less flexible. If you want to extend the record and get that data in from the outside, well, then you have to extend the service to be able to get the data in. You'd have to do this either way, DTO or method signature.

I also thought about handing in the job record instead of the `ActivateJobsRequest` but it didn't have all the fields we need.

Reply:
Yeah, I think the current solution is the best one. 👍